fmf_scheduler.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python3
+"""
+FMF scheduler for nosetests
+Found test via fmf files and creates dynamic testclassed based on that.
+"""
+
+
+import os
+import sys
+import imp
+import logging
+import unittest
+
+from colin.core.target import Target, is_compatible
+from colin.core.checks.fmf_check import ExtendedTree, FMFAbstractCheck
+from colin.core.constant import PASSED
+from colin.core.ruleset.ruleset import get_checks_path
+
+from colin.core.checks.dockerfile import DockerfileAbstractCheck, DockerfileLabelAbstractCheck,\
+    InstructionCountAbstractCheck, InstructionAbstractCheck
+
+CLASS_MAPPING = {"DockerfileAbstractCheck": DockerfileAbstractCheck,
+                 "DockerfileLabelAbstractCheck": DockerfileLabelAbstractCheck,
+                 "InstructionCountAbstractCheck": InstructionCountAbstractCheck,
+                 "InstructionAbstractCheck": InstructionAbstractCheck
+                 }
+logger = logging.getLogger(__name__)
+
+
+def get_log_level():
+    return int(os.environ.get(("DEBUG")) or logging.INFO)
+
+
+# COPYied from:
+# https://eli.thegreenplace.net/2014/04/02/dynamically-generating-python-test-cases
+class DynamicClassBase(unittest.TestCase):
+    """
+    Basic Derived test Class
+    """
+    backendclass = None
+    longMessage = True
+
+
+def make_check_function(target):
+    """
+    rename check function to test, to be able to find this function via testing frameworks
+    :return: function
+    """
+
+    def test(self):
+        out = self.backendclass().check(target)
+        if out.status is not PASSED:
+            raise AssertionError("test:{} -> {}".format(
+                self.backendclass.name,
+                str(out))
+            )
+    return test
+
+
+def make_base_fmf_class_abstract(node, target, base_class=DynamicClassBase, fmf_class=FMFAbstractCheck):
+    class_name = node.data["class"]
+    out_class_name = node.name.rsplit("/", 1)[-1]
+    outclass = type(out_class_name, (base_class,), {
+        'test': make_check_function(target=target),
+        'backendclass': type(class_name, (fmf_class, CLASS_MAPPING[class_name],), {
+            '__doc__': node.data.get("description"),
+            'name': out_class_name,
+            'metadata': node,
+        })
+    })
+    return outclass
+
+
+def make_wrapped_fmf_class(node, target):
```

Comment:
Function is a bit complicated. What about to add an example into docstring for better understanding.